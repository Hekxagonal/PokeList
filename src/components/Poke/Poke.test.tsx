import { screen } from '@testing-library/react';
import Component from './index';
import renderTheme from '../../global/theme/render-theme';
import formatName from './format';

describe('<PokeProfile />', () => {
  const defaultValues = {
    name: 'glaceon',
    types: ['ice', 'fire'],
    sprites: {
      front_default: 'frontURL',
      back_shiny: 'backURL',
    },
    anim: 1,
  };

  it('format func should be working', () => {
    const pikachu = formatName('pikachu');
    const nidoranF = formatName('nidoran-f');
    const nidoranM = formatName('nidoran-m');
    const eeveeE = formatName('eevee-e');

    expect(pikachu).toBe('Pikachu');
    expect(nidoranF).toBe('Nidoran♂');
    expect(nidoranM).toBe('Nidoran♀');
    expect(eeveeE).toBe('Eevee');
  });
  it('should be a processed name', () => {
    renderTheme(<Component {...defaultValues} />);

    const heading = screen.getByRole('heading', {
      name: formatName(defaultValues.name),
    });
    expect(heading).toBeInTheDocument();
  });
  it('should be two types', () => {
    renderTheme(<Component {...defaultValues} />);

    const iceType = screen.getByText('ICE');
    const fireType = screen.getByText('FIRE');
    expect(iceType).toBeInTheDocument();
    expect(fireType).toBeInTheDocument();
  });
  it('should be only a type', () => {
    const values = { ...defaultValues };
    values.types.pop();
    renderTheme(<Component {...values} />);

    const iceType = screen.getByText('ICE');
    const fireType = screen.queryByText('FIRE');
    expect(iceType).toBeInTheDocument();
    expect(fireType).toBe(null);
  });

  it('should match snapshot', () => {
    const view = renderTheme(<Component {...defaultValues} />);

    expect(view).toMatchSnapshot();
  });
});
