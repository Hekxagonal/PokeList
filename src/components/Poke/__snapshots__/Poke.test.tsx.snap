// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokeProfile /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  box-shadow: 0px 0px 100px rgba(0,0,0,0.2);
  border-radius: 10px;
  padding: 25px 10px 10px 10px;
  margin-top: 60px;
  width: 195px;
  height: 70px;
  -webkit-animation: 0.1s ease-out 0s 1 diRwIN;
  animation: 0.1s ease-out 0s 1 diRwIN;
  background-color: white;
}

.c3 {
  font-size: 15px;
  color: rgba(0,0,0,1);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 50%;
}

.c5 {
  padding: 4px;
  border-radius: 3px;
  font-size: 12px;
  color: white;
  font-family: "Inconsolata",monospace;
  font-weight: bold;
  background: rgba(149,181,237,1);
}

.c1 {
  position: absolute;
  margin-top: -70px;
  margin-left: -20px;
  z-index: 1;
  -webkit-animation: 2s ease-out 0s 1 hfyfRi;
  animation: 2s ease-out 0s 1 hfyfRi;
}

.c2 {
  margin-top: -70px;
  margin-left: 20px;
  position: absolute;
  -webkit-filter: blur(1px);
  filter: blur(1px);
}

<body>
    <div>
      <div
        class="c0"
      >
        <img
          class="c1"
          src="frontURL"
        />
        <img
          class="c2"
          src="backURL"
        />
        <h1
          class="c3"
        >
          Glaceon
        </h1>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            ICE
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  background: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  box-shadow: 0px 0px 100px rgba(0,0,0,0.2);
  border-radius: 10px;
  padding: 25px 10px 10px 10px;
  margin-top: 60px;
  width: 195px;
  height: 70px;
  -webkit-animation: 0.1s ease-out 0s 1 diRwIN;
  animation: 0.1s ease-out 0s 1 diRwIN;
  background-color: white;
}

.c3 {
  font-size: 15px;
  color: rgba(0,0,0,1);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 50%;
}

.c5 {
  padding: 4px;
  border-radius: 3px;
  font-size: 12px;
  color: white;
  font-family: "Inconsolata",monospace;
  font-weight: bold;
  background: rgba(149,181,237,1);
}

.c1 {
  position: absolute;
  margin-top: -70px;
  margin-left: -20px;
  z-index: 1;
  -webkit-animation: 2s ease-out 0s 1 hfyfRi;
  animation: 2s ease-out 0s 1 hfyfRi;
}

.c2 {
  margin-top: -70px;
  margin-left: 20px;
  position: absolute;
  -webkit-filter: blur(1px);
  filter: blur(1px);
}

<div>
    <div
      class="c0"
    >
      <img
        class="c1"
        src="frontURL"
      />
      <img
        class="c2"
        src="backURL"
      />
      <h1
        class="c3"
      >
        Glaceon
      </h1>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          ICE
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
